src/lib.rs

```diff
@@ -0,0 +1,261 @@
+use std::{time::{SystemTime, UNIX_EPOCH}, sync::{Mutex, mpsc::{Receiver, channel}}};
+
+use discord_presence::Client;
+use rrplug::prelude::*;
+
+const EMPTY: String = String::new();
+
+static mut WAS_IN_GAME: bool = false;
+static mut RESET_SINGLE_PLAYER_TIMER: bool = true;
+
+struct ActivityData {
+    party: (u32, u32),
+    details: String,
+    state: String,
+    large_image: String,
+    large_text: String,
+    small_image: String,
+    small_text: String,
+    end: Option<u64>,
+    start: Option<u64>,
+}
+
+struct DiscordRpc {
+    gamestate: Option<GameState>,
+    serverinfo: Option<ServerInfo>,
+    drpc: Option<Mutex<Client>>,
+    recv: Option<Receiver<bool>>
+}
+
+impl Plugin for DiscordRpc {
+    fn new() -> Self {
+        Self {
+            gamestate: None,
+            serverinfo: None,
+            drpc: None,
+            recv: None
+        }
+    }
+
+    fn initialize(&mut self, external_plugin_data: ExternalPluginData) {
+        self.gamestate = external_plugin_data.get_game_state_struct();
+        self.serverinfo = external_plugin_data.get_server_info_struct();
+        println!("[DiscordRPC] discord rpc plugin initialized");
+
+        let mut drpc = Client::new(941428101429231617);
+        
+        let _ = drpc.start();
+
+        let (send, recv) = channel::<bool>();
+        self.recv = Some(recv);
+        let send = Mutex::new(send);
+
+        // drpc.block_until_event(discord_presence::Event::Ready).unwrap(); // hmm doesn't work?
+
+        drpc.on_ready(move |_| send.lock().unwrap().send(true).unwrap());
+
+        self.drpc = Some(Mutex::new(drpc));
+    }
+
+    fn main(&self) {
+        let gamestate = self.gamestate.as_ref().unwrap();
+        let serverinfo = self.serverinfo.as_ref().unwrap();
+        let mut drpc = self.drpc.as_ref().unwrap().lock().unwrap();
+        let recv = self.recv.as_ref().unwrap();
+        
+        println!("[DiscordRPC] waiting for discord rpc to be ready");
+        loop {
+            if recv.try_recv().is_ok() {
+                println!("[DiscordRPC] discord rpc initialized");
+                break;
+            }
+            wait(1000)
+        }
+
+        loop {
+            if let Err(err) = drpc.clear_activity() {
+                #[cfg(debug_assertions)]
+                println!("[DiscordRPC] coudln't clear activity because of {:?}", err)
```

Comment:
```suggestion
                println!("[DiscordRPC] couldn't clear activity because of {:?}", err)
```

Typo that I missed when last looking at it ^^